### PR TITLE
✂️ dependency: remove puppeteer in dependencies

### PR DESCRIPTION
**Pull request by chenshuai2144**
* 不需要默认安装 puppeteer
* 自动使用用户的 chrome，如果找不到会再安装  puppeteer

### CODE REVIEW

File: src/e2e/getBrowser.js [hunk 1]
```diff
@@ -1,7 +1,44 @@
...
+      stdio: 'inherit',
+    });
+    testCmd.on('exit', (code) => {
+      resolve(code);
+    });
+  });
+};
 
 const getBrowser = async () => {
+  try {
```
Comment (afc163) [inline src/e2e/getBrowser.js]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/manual/afc163/ant-design-pro/pullRequest?prid=5242326)

File: src/e2e/getBrowser.js [hunk 2]
```diff
@@ -1,7 +1,45 @@
-import puppeteer from 'puppeteer';
+/* eslint-disable global-require */
+/* eslint-disable import/no-extraneous-dependencies */
+const { spawn } = require('child_process');
+const findChrome = require('carlo/lib/find_chrome');
+
+const installPuppeteer = () => {
+  return new Promise((resolve) => {
+    const testCmd = spawn(/^win/.test(process.platform) ? 'npm.cmd' : 'npm', ['i', 'puppeteer'], {
```
Comment (afc163) [inline src/e2e/getBrowser.js]: 用户有可能用 `yarn` `cnpm` 等。

Comment (chenshuai2144) [reply src/e2e/getBrowser.js]: 用户不会跑到这里的，这个是给ci 留着的

应该不会有开发者电脑上没装 chrome 吧 🙈🙉🙊

Comment (afc163) [reply src/e2e/getBrowser.js]: 用户是不是也会本地跑跑测试？

Comment (chenshuai2144) [reply src/e2e/getBrowser.js]: 本地会走 find chrome，走自己本地的 chrome ，这样就不用安装 puppeteer

Comment (afc163) [reply src/e2e/getBrowser.js]: ci 也可能用 yarn 或者 tnpm。

Comment (chenshuai2144) [reply src/e2e/getBrowser.js]: 对于 ci 用什么都无所谓的吧，
封装到 umi 中的时候可以做成通用的 支持 npmClient

Comment (afc163) [reply src/e2e/getBrowser.js]: 这个代码 push 到 master 就可能被拿到各个公司环境下跑 ci 的。

Comment (chenshuai2144) [reply src/e2e/getBrowser.js]: 我来想个办法

File: tests/getBrowser.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/* eslint-disable global-require */
+/* eslint-disable import/no-extraneous-dependencies */
+const findChrome = require('carlo/lib/find_chrome');
+
+const getBrowser = async () => {
+  try {
```
Comment (afc163) [inline tests/getBrowser.js]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/manual/afc163/ant-design-pro/pullRequest?prid=5242326)

File: tests/beforeTest.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    // eslint-disable-next-line import/no-unresolved
+    const findChromePath = await findChrome({});
+    const { executablePath } = findChromePath;
+    console.log(`🧲 find you browser in ${executablePath}`);
+    return;
+  } catch (error) {
+    console.log('🧲 no find chrome');
+  }
+
+  try {
```
Comment (afc163) [inline tests/beforeTest.js]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/manual/afc163/ant-design-pro/pullRequest?prid=5242326)

File: tests/PuppeteerEnvironment.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  async setup() {
+    const browser = await getBrowser();
+    const page = await browser.newPage();
+    this.global.browser = browser;
+    this.global.page = page;
+  }
+
+  async teardown() {
+    const { page, browser } = this.global;
```
Comment (afc163) [inline tests/PuppeteerEnvironment.js]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/manual/afc163/ant-design-pro/pullRequest?prid=5242326)

File: tests/beforeTest.js (hunk 4 above)
Comment (afc163) [inline tests/beforeTest.js]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/manual/afc163/ant-design-pro/pullRequest?prid=5242326)

File: tests/PuppeteerEnvironment.js (hunk 5 above)
Comment (afc163) [inline tests/PuppeteerEnvironment.js]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/manual/afc163/ant-design-pro/pullRequest?prid=5242326)

File: tests/getBrowser.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
```
Comment (afc163) [inline tests/getBrowser.js]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/manual/afc163/ant-design-pro/pullRequest?prid=5242326)

Comment (afc163) [inline tests/getBrowser.js]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/manual/afc163/ant-design-pro/pullRequest?prid=5242326)

File: tests/beforeTest.js (hunk 4 above)
Comment (afc163) [inline tests/beforeTest.js]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/manual/afc163/ant-design-pro/pullRequest?prid=5242326)

File: tests/PuppeteerEnvironment.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
```
Comment (afc163) [inline tests/PuppeteerEnvironment.js]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/manual/afc163/ant-design-pro/pullRequest?prid=5242326)

**Pull request merged**